Loop/Views/SuspendThresholdEditor.swift

```diff
@@ -0,0 +1,140 @@
+//
+//  SuspendThresholdEditor.swift
+//  Loop
+//
+//  Created by Michael Pangburn on 4/10/20.
+//  Copyright © 2020 LoopKit Authors. All rights reserved.
+//
+
+import SwiftUI
+import HealthKit
+import LoopKit
+import LoopKitUI
+
+
+struct SuspendThresholdEditor: View {
+    @State private var value = HKQuantity(unit: .milligramsPerDeciliter, doubleValue: 80)
+    @State private var isEditing = false
+    @State private var showingConfirmationAlert = false
+
+    private let originalValue: HKQuantity?
+    private let unit: HKUnit
+    private let save: (_ suspendThreshold: HKQuantity) -> Void
+
+    @Environment(\.dismiss) var dismiss
+
+    init(value: HKQuantity?, unit: HKUnit, onSave save: @escaping (_ suspendThreshold: HKQuantity) -> Void) {
+        self._value = State(initialValue: value ?? Self.defaultValue(for: unit))
+        self.originalValue = value
+        self.unit = unit
+        self.save = save
+    }
+
+    private static func defaultValue(for unit: HKUnit) -> HKQuantity {
+        switch unit {
+        case .milligramsPerDeciliter:
+            return HKQuantity(unit: .milligramsPerDeciliter, doubleValue: 80)
+        case .millimolesPerLiter:
+            return HKQuantity(unit: .millimolesPerLiter, doubleValue: 4.5)
+        default:
+            fatalError("Unsupported glucose unit \(unit)")
+        }
+    }
+
+    var body: some View {
+        ConfigurationPage(
+            title: Text("Suspend Threshold"),
+            isSaveButtonEnabled: isSaveButtonEnabled,
+            cards: {
+                // TODO: Remove conditional when Swift 5.3 ships
+                // https://bugs.swift.org/browse/SR-11628
+                if true {
+                    Card {
+                        SuspendThresholdPicker(value: $value, unit: unit, isEditing: $isEditing)
+                        SuspendThresholdDescription()
+                    }
+                }
+            },
+            actionAreaContent: {
+                if warningThreshold != nil {
+                    SuspendThresholdGuardrailWarning(safetyClassificationThreshold: warningThreshold!)
+                        .padding(.horizontal)
+                        .transition(AnyTransition.opacity.combined(with: .move(edge: .bottom)))
+                }
+            },
+            onSave: {
+                if self.warningThreshold == nil {
+                    self.saveAndDismiss()
+                } else {
+                    self.showingConfirmationAlert = true
+                }
+            }
+        )
+        .alert(isPresented: $showingConfirmationAlert, content: confirmationAlert)
```

Comment:
Very nice! Looking forward to bringing it into the Dexcom UI.

Question about the `ConfigurationPage`. Does the action always have to be in the context of `Save`? If not, what do you think about allowing the copy for the action button to be overridden?

Reply:
Doesn't need to be constrainted to 'Save'. I can generalize this is a follow-up LoopKit PR if that works for you.

Reply:
I'm good with that if you think it makes sense to make the action context more generic and agree there are potential use cases.

Reply:
Totally—there's no reason that this general UI structure needs to be tied to 'save' actions.
It's easy enough to add a convenience initializer for the 'save' case—so I could make the change without even disrupting this call site.